fecgov/openFEC
# Add first_f1_date and filter to /totals/by_entity/

## Summary (required)

- Resolves #4951 

Add first_f1_date and min/max_first_f1_date filter to /totals/by_entity/.  
Endpoint /committee/{committee_id}/totals/ will have first_f1_date since it shares the same data resource .

### Required reviewers

Two developers required.

## Impacted areas of the application

/totals/by_entity/ endpoint
/committee/{committee_id}/totals/

## How to test

- Check out this branch
- Test flyway migration:  after migration, new column `first_f1_date` should be seen in these mv/views:
```
ofec_totals_combined_mv
ofec_totals_combined_vw
ofec_committee_totals_per_cycle_vw
ofec_totals_pac_party_vw
ofec_totals_house_senate_mv
ofec_totals_ie_only_mv
```
- Test endpoint:  make changes for testing data and start api on local
```
webservices/common/models/totals.py:

class CommitteeTotalsPacParty(CommitteeTotals):
    __tablename__ = 'ofec_totals_pac_party_vw_tmp_hc'

class CommitteeTotalsHouseSenate(CommitteeTotals):
    __tablename__ = 'ofec_totals_house_senate_mv_tmp_hc'

class CommitteeTotalsPerCycle(CommitteeTotals):
    __tablename__ = 'ofec_committee_totals_per_cycle_vw_tmp_hc'
```

http://127.0.0.1:5000/v1/totals/pac-party/?sort_nulls_last=false&sort_hide_null=false&page=1&sort_null_only=false&sort=-cycle&min_first_f1_date=1-1-2017&per_page=20&max_first_f1_date=12-31-2017

http://127.0.0.1:5000/v1/totals/house-senate/?min_first_f1_date=2021-01-01&sort_null_only=false&page=1&sort=-cycle&sort_hide_null=false&per_page=20&sort_nulls_last=false&max_first_f1_date=2021-12-31

http://127.0.0.1:5000/v1/totals/presidential/?min_first_f1_date=2021-01-01&sort_null_only=false&page=1&sort=-cycle&sort_hide_null=false&per_page=20&sort_nulls_last=false&max_first_f1_date=2021-12-31
